### PR TITLE
Fix for "Application failed to start after update" when an external network is on a watched service

### PR DESCRIPTION
**Pull request by kimdcottrell**
**What I did**

I'll keep editing this part with progress. WIP

At present:
- e2e test that shows bug reported in issue
- fix for bug :partying_face: 
- rebased commits so I could amend a signoff on work done

Remaining:
- duplicate code in e2e test should be refactored.
- anything that gets flagged by automated tests below

**Related issue**
Fixes: https://github.com/docker/compose/issues/11081

**(not mandatory) A picture of a cute animal, if possible in relation to what you did**
https://giphy.com/gifs/justviralnet-funny-dog-corgi-ZFFVNwIbjsKtP3lHYK

### CODE REVIEW

**Comment by kimdcottrell** [comment]
The test right now has some duplicate code from elsewhere in the same file. I'm presently debating if frameworks.go should contain the refactored logic around running `watch` and killing the processes during cleanup.

**Comment by kimdcottrell** [comment]
Bug issue solved! :D 

I'll tackle the refactoring later, along with any possibly failing tests and suggestions.

**Comment by glours** [comment]
Hey @kimdcottrell 
Compose policy enforce you to sign-up your commits, can you follow [the steps described here](https://github.com/docker/compose/pull/11092/checks?check_run_id=17688468504) to pass the DCO check please?

File: pkg/compose/create.go [hunk 1]
```diff
@@ -1124,13 +1124,26 @@ func (s *composeService) resolveExternalNetwork(ctx context.Context, n *types.Ne
 	networks, err := s.apiClient().NetworkList(ctx, moby.NetworkListOptions{
 		Filters: filters.NewArgs(filters.Arg("name", n.Name)),
 	})
+
 	if err != nil {
 		return err
 	}
 
+	if len(networks) == 0 {
+		networks, err = s.apiClient().NetworkList(ctx, moby.NetworkListOptions{
```
Comment (ndeloof) [inline pkg/compose/create.go:1133]: here we assume n.Name is actually network ID, then `NetworkInspect` would be a better API to use

Comment (kimdcottrell) [reply pkg/compose/create.go:1133]: Thank you!

Comment (kimdcottrell) [reply pkg/compose/create.go]: This has been changed.

**Review by ndeloof — APPROVED** [review]

**Comment by kimdcottrell** [comment]
I started on a possibly iffy refactor of the duplicate code between the tests (the docker compose watch command and the cleanup function), though I see this PR has already been approved. I'll stop working on that and assume no more work is required at this time, though I'm happy to come back to this if anything comes up. :)

**Comment by kimdcottrell** [comment]
Fixed some outstanding linter problems and remerged the base branch incase that was holding back the merge of this branch.

**Pull request merged**